cli/src/deno.rs

```diff
@@ -81,9 +83,45 @@ pub async fn run(extension_state: ExtensionState, entry_point: &str) -> Result<(
 }
 
 /// See https://github.com/denoland/deno/blob/main/core/examples/ts_module_loader.rs.
-struct TypescriptModuleLoader;
+struct ExtensionsModuleLoader;
+
+impl ExtensionsModuleLoader {
+    async fn load_from_filesystem(path: &Url) -> Result<String> {
+        let path = path
+            .to_file_path()
+            .map_err(|_| anyhow!("{path:?}: is not a path"))?;
+
+        let extensions_path = extensions_path()?;
+        if !path.starts_with(&extensions_path) {
+            return Err(anyhow!(
+                "`{}`: importing from paths outside of the extension's directory is not allowed",
+                path.to_string_lossy(),
+            ));
+        }
```

Comment:
Do you have an idea how we could cleanly prevent users from specifying dependencies between extensions here? I feel like we should better be safe than sorry and not allow this.